plato/draw/Ellipsoids.py

```diff
@@ -0,0 +1,27 @@
+import itertools
+import numpy as np
+from .internal import Shape, ShapeDecorator, ShapeAttribute
+
+@ShapeDecorator
+class Ellipsoids(Shape):
+    """A collection of ellipsoids with identical dimensions.
+
+    Each ellipsoid can have its own position, orientation, and
+    color. All shapes drawn by this primitive share common axis
+    lengths.
+    """
+
+    _ATTRIBUTES = list(itertools.starmap(ShapeAttribute, [
+        ('positions', np.float32, (0, 0, 0), 2, True,
+         'Position of each particle'),
+        ('orientations', np.float32, (1, 0, 0, 0), 2, True,
+         'Orientation quaternion of each particle'),
+        ('colors', np.float32, (.5, .5, .5, 1), 2, True,
+         'Color, RGBA, [0, 1] for each particle'),
+        ('a', np.float32, .5, 0, False,
+         'Half-length in the x-direction'),
```

Comment:
Could we call it "radius" instead?

Reply:
The standard language for this is semi-axes I believe. But I think any variable name is fine as long as it's well documented. Radius may be easier for a coder to guess even if it's not "correct" in the strict sense of the word.